lib/auth/clt.go

```diff
@@ -2670,6 +2650,73 @@ func (c *Client) Ping(ctx context.Context) (proto.PingResponse, error) {
 	return *rsp, nil
 }
 
+// ClientServices implements IClientServices interface.
+type ClientServices struct {
+	roundtrip.Client
+}
+
+// PostJSON is a generic method that issues http POST request to the server
+func (c *ClientServices) PostJSON(
+	endpoint string, val interface{}) (*roundtrip.Response, error) {
+	return httplib.ConvertResponse(c.Client.PostJSON(context.TODO(), endpoint, val))
+}
+
+// PutJSON is a generic method that issues http PUT request to the server
+func (c *ClientServices) PutJSON(
+	endpoint string, val interface{}) (*roundtrip.Response, error) {
+	return httplib.ConvertResponse(c.Client.PutJSON(context.TODO(), endpoint, val))
+}
+
+// PostForm is a generic method that issues http POST request to the server
+func (c *ClientServices) PostForm(
+	endpoint string,
+	vals url.Values,
+	files ...roundtrip.File) (*roundtrip.Response, error) {
+	return httplib.ConvertResponse(c.Client.PostForm(context.TODO(), endpoint, vals, files...))
+}
+
+// Get issues http GET request to the server
+func (c *ClientServices) Get(u string, params url.Values) (*roundtrip.Response, error) {
+	return httplib.ConvertResponse(c.Client.Get(context.TODO(), u, params))
+}
+
+// Delete issues http Delete Request to the server
+func (c *ClientServices) Delete(u string) (*roundtrip.Response, error) {
+	return httplib.ConvertResponse(c.Client.Delete(context.TODO(), u))
+}
+
+// Endpoint returns a URL constructed from parts and version appended, e.g.
+func (c *ClientServices) Endpoint(params ...string) string {
+	return c.Client.Endpoint(params...)
+}
+
+// PostSessionSlice allows clients to submit session stream chunks to the audit log
+// (part of evets.IAuditLog interface)
+//
+// The data is POSTed to HTTP server as a simple binary body (no encodings of any
+// kind are needed)
+func (c *ClientServices) PostSessionSlice(slice events.SessionSlice) error {
+	data, err := slice.Marshal()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	r, err := http.NewRequest("POST", c.Endpoint("namespaces", slice.Namespace, "sessions", slice.SessionID, "slice"), bytes.NewReader(data))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	r.Header.Set("Content-Type", "application/grpc")
```

Comment:
Hmm, why is this `application/grpc`?
If it's actually gRPC, should it go over the gRPC transport?

Reply:
@klizhentas do you recall why you had to explicitly set these headers?